### DISCUSSION TITLE
Modify Subscribed ics files

### DISCUSSION THREAD

**Discussion by varuniyer** [bitfireAT/icsx5#28] [Ideas]
Would it be possible to edit events in ics files that we've subscribed to? I understand this app has a different goal than the DAVx5 app, but it would be nice to be able to modify and sync to locally stored ics files.

**Comment by rfc2822** [comment]
At the moment, the scope of ICSx⁵ is only to subscribe to read-only Webcal feeds.

What would your use-case be? There may be other methods that suit better.

**Reply by varuniyer** [reply]
I would like to have an ics file encrypted locally and then synced to an external server using rclone. I guess I'm looking for something like a CalDAV server with client-side encryption. The closest thing to this is Etesync, but their Django backend code hasn't been updated in several months beyond simple README changes.

**Reply by rfc2822** [reply]
I see… CalDAV is not suitable for client-side encryption. It would however be possible over WebDAV, and I have already thought about that because it would be a cool thing. However this has the size of an own project and is currently not within our scope (too much other things)…

**Reply by devycarol** [reply]
For me, I just think that time is long overdue for there to be a solution to have your Android calendars interact directly with locally stored files rather than being reliant on external services to have a cross-synced calendar. Ideally this would be put on the calendar applications themselves, but principally I want to be able to use Syncthing to synchronize my calendars between devices without needing to use middle-man tools like DecSync and self-hosted CalDAV servers.

Basically, it's important to me that the shared data be in standardized, non-obfuscated form. Correct me if I'm wrong, but I believe that ICS is the closest thing there is to that for calendar storage. Most desktop calendar applications support read/write ICS calendaring, and I would appreciate if the same paradigm came to be supported on Android.

**Reply by rfc2822** [reply]
Yes, such a sync adapter between local iCalendar files and calendar entries could surely be made. Our library [ical4android](https://github.com/bitfireAT/ical4android) would already provide many functions for that. However this feature doesn't fall in the scope of DAVx⁵ and should be a separate app.
